contracts/Enigma.sol

```diff
@@ -191,7 +247,7 @@ contract Enigma is EnigmaStorage, EnigmaEvents, Getters, Ownable {
     function deposit(address _custodian, uint _amount)
     public
     isUpdatedEnigmaContract
-    workerRegistered(_custodian)
+    workerRegistered(state.stakingToOperatingAddresses[_custodian])
```

Comment:
Could you please elaborate on this logic?
From the requirements: the staking is done form the staking address and only afterwards the `setOperatingAddress` can be called. But here it seems that in the time of the `deposit` the operating address is already known...

Reply:
Staking is only done from operating address, but we ensure that `setOperatingAddress` is called first from the staking address. We need this order of operations because the workers mapping has a key of operating address. This includes a struct with a `balance` field, which is adjusted when a staking address deposits. In this way, staking and operating are sort of independent but also dependent. They are independent in the sense that a staking address can withdraw its deposit from an operating address, and set its operating address to a different worker, and redeposit. It is dependent in the sense that, from a technical implementation standpoint, the cleanest way is to set its balance as a value in the workers mapping where the key is the operating address, and we maintain an additional staking address -> operating address mapping. happy to hop on a call to clear things up if need be!

Reply:
Just to be clear, in the first sentence `Staking is only done from operating address`, you mean `Register is only done from operating address`, right?
Agree with all the rest, thanks

Reply:
oops yes!! register is done from operating address :)